ouroboros-network/src/Ouroboros/Network/PeerSelection/PeerSelectionActions.hs

```diff
@@ -120,34 +134,41 @@ withPeerSelectionActions
         localRootsVar)
       (\thread -> k thread peerSelectionActions)
   where
-    -- We first try to get public root peers from the ledger, but if it fails
-    -- (for example because the node hasn't synced far enough) we fall back
-    -- to using the manually configured bootstrap root peers.
-    requestPublicRootPeers :: Int -> m (Map peeraddr (PeerAdvertise, IsLedgerPeer), DiffTime)
-    requestPublicRootPeers n = do
-      peers_m <- getLedgerPeers (NumberOfPeers $ fromIntegral n) AllLedgerPeers
-      case peers_m of
-           -- No peers from Ledger
-           Nothing    -> do
-             (m, dt) <- requestConfiguredRootPeers n
-             let m' = Map.map (\a -> (a, IsNotLedgerPeer)) m
-             return (m', dt)
+    -- We start by reading the current ledger state judgement, if it is
+    -- YoungEnough we only care about fetching for ledger peers, otherwise we
+    -- aim to fetch bootstrap peers.
+    requestPublicRootPeers :: LedgerPeersKind -> Int -> m (PublicRootPeers peeraddr, DiffTime)
+    requestPublicRootPeers ledgerPeersKind n = do
```

Comment:
What if bootstrap peers are off? I think in that case we should keep the current logic (which is more compatible with genesis).

Reply:
You're right! If bootstrap peers are off we should just do what the `YoungEnough` is doing